snapcraft/plugins/jhbuild.py

```diff
@@ -0,0 +1,773 @@
+# -*- Mode:Python; indent-tabs-mode:nil; tab-width:4 -*-
+#
+# Copyright (C) 2016 Canonical Ltd
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License version 3 as
+# published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+"""The JHBuild plugin is used for GNOME-based parts.
+
+JHBuild is a tool used to build components of the GNOME ecosystem from version
+control. See the JHBuild manual for more details:
+https://developer.gnome.org/jhbuild/stable/.
+
+The plugin can be customized with the following keys:
+
+  - modules: (required) a list of modules to include in the part, modules that
+             should be skipped should be prefixed with '-', e.g. '-WebKit'
+  - module-set: (default: gnome-world) the module set JHBuild should use
+  - module-set-dir: the directory where the module set is located
+  - snap-revision: the revision of the snap this part is included in
```

Comment:
You mean the store can potentially overwrite the revision of the uploaded snap to something other than what the developer specified in the snapcraft.yaml? That's really a big problem for this commit then because it's making a pretty big assumption that the location of /snap/<snap-name>/<snap-revision> is fixed.